### ISSUE TITLE
BigNumber Error when currencies specified in pathfinder

### ISSUE THREAD

**Issue by codedot** [ripple/ripple-lib#737]
When specifying `source.currencies[]` in the `pathfind` argument of `getPaths()`, a `BigNumber Error` arises. This is due to the `requestPathFind()` routine [calling](https://github.com/ripple/ripple-lib/blob/50c6af31589b6afd9508403c52000257a40d760c/src/ledger/pathfind.js#L42) `toRippledAmount()` on currencies instead of amounts. The `toRippledAmount()` assumes its argument to have a value property, thus leading to that `BigNumber Error` when [calling](https://github.com/ripple/ripple-lib/blob/50c6af31589b6afd9508403c52000257a40d760c/src/common/utils.js#L28) `xrpToDrops()` on `undefined`.

**Comment by intelliot** [comment]
@codedot Thanks for your patience! We've made several releases of ripple-lib since this report. Could you confirm that it's still an issue in 0.17.8, and provide a code snippet that demonstrates the error?

Thank you!

**Comment by codedot** [comment]
```
$ npm install ripple-lib
/tmp/tmp.fXvsXGbxcf
└─┬ ripple-lib@0.17.8 
  ├─┬ babel-polyfill@6.26.0 
  │ ├── core-js@2.5.1 
  │ └── regenerator-runtime@0.10.5 
  ├─┬ babel-runtime@6.26.0 
  │ └── regenerator-runtime@0.11.0 
  ├── bignumber.js@4.1.0 
  ├─┬ https-proxy-agent@1.0.0 
  │ ├─┬ agent-base@2.1.1 
  │ │ └── semver@5.0.3 
  │ ├─┬ debug@2.6.9 
  │ │ └── ms@2.0.0 
  │ └── extend@3.0.1 
  ├── jsonschema@1.2.0 
  ├── lodash@4.17.4 
  ├─┬ ripple-address-codec@2.0.1 
  │ ├─┬ hash.js@1.1.3 
  │ │ └── minimalistic-assert@1.0.0 
  │ └─┬ x-address-codec@0.7.2 
  │   └── base-x@1.1.0 
  ├─┬ ripple-binary-codec@0.1.11 
  │ ├── bn.js@4.11.8 
  │ ├─┬ create-hash@1.1.3 
  │ │ ├─┬ cipher-base@1.0.4 
  │ │ │ └── safe-buffer@5.1.1 
  │ │ ├─┬ ripemd160@2.0.1 
  │ │ │ └── hash-base@2.0.2 
  │ │ └── sha.js@2.4.9 
  │ ├── decimal.js@5.0.8 
  │ └── inherits@2.0.3 
  ├── ripple-hashes@0.3.1 
  ├─┬ ripple-keypairs@0.10.0 
  │ ├─┬ babel-runtime@5.8.38 
  │ │ └── core-js@1.2.7 
  │ ├── bn.js@3.3.0 
  │ ├── brorand@1.1.0 
  │ └─┬ elliptic@5.2.1 
  │   └── bn.js@3.3.0 
  ├── ripple-lib-transactionparser@0.6.2 
  └─┬ ws@1.1.5 
    ├── options@0.0.6 
    └── ultron@1.0.2 

npm WARN enoent ENOENT: no such file or directory, open '/tmp/tmp.fXvsXGbxcf/package.json'
npm WARN tmp.fXvsXGbxcf No description
npm WARN tmp.fXvsXGbxcf No repository field.
npm WARN tmp.fXvsXGbxcf No README data
npm WARN tmp.fXvsXGbxcf No license field.
$ cat >test.js
const ripple = require("ripple-lib");
const api = new ripple.RippleAPI({
        server: "wss://s1.ripple.com"
});
const root = "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh";

api.connect().then(() => api.getPaths({
        source: {
                address: root,
                currencies: [{
                        currency: "XRP"
                }]
        },
        destination: {
                address: root,
                amount: {
                        value: "1",
                        currency: "XRP"
                }
        }
})).then(list => {
        console.info(list);
        process.exit();
}).catch(error => {
        console.error(error);
        process.exit(1);
});
$ node test.js
{ BigNumber Error: new BigNumber() not a number: undefined
    at raise (/tmp/tmp.fXvsXGbxcf/node_modules/bignumber.js/bignumber.js:1190:25)
    at /tmp/tmp.fXvsXGbxcf/node_modules/bignumber.js/bignumber.js:1178:33
    at new BigNumber (/tmp/tmp.fXvsXGbxcf/node_modules/bignumber.js/bignumber.js:193:67)
    at xrpToDrops (/tmp/tmp.fXvsXGbxcf/node_modules/ripple-lib/dist/npm/common/utils.js:25:10)
    at toRippledAmount (/tmp/tmp.fXvsXGbxcf/node_modules/ripple-lib/dist/npm/common/utils.js:30:12)
    at /tmp/tmp.fXvsXGbxcf/node_modules/ripple-lib/dist/npm/ledger/pathfind.js:42:21
    at Array.map (native)
    at requestPathFind (/tmp/tmp.fXvsXGbxcf/node_modules/ripple-lib/dist/npm/ledger/pathfind.js:41:60)
    at RippleAPI.getPaths (/tmp/tmp.fXvsXGbxcf/node_modules/ripple-lib/dist/npm/ledger/pathfind.js:114:10)
    at api.connect.then (/tmp/tmp.fXvsXGbxcf/test.js:7:30) name: 'BigNumber Error' }
$ 
```

**Comment by FKSRipple** [comment]
this has been solved in #1074 (repro above is no longer failing)
